.eleventy.js

```diff
@@ -3,13 +3,32 @@ require('ts-node').register({
   project: 'tsconfig.json',
 })
 
+const path = require('node:path')
+const { createElement, StrictMode } = require('react')
+const { renderToString } = require('react-dom/server')
+
 const { createNunjucksEnvironment } = require('./src')
 
 module.exports = (config) => {
   config.setLibrary('njk', createNunjucksEnvironment())
 
   config.addPassthroughCopy({ dist: '.' })
 
+  config.addWatchTarget(path.join(__dirname, 'src/react'))
+  config.addShortcode('react', async (componentPath) => {
+    /* The eslint rule import/no-dynamic-require is asking for
+    * await import() however this has been difficult to get this working the ts-node
+    * https://github.com/TypeStrong/ts-node/issues/1548
+    */
+    // eslint-disable-next-line import/no-dynamic-require, global-require
+    const component = require(path.join(__dirname, 'src/react', componentPath)).default
+    return renderToString(
+      createElement(StrictMode, {
+        children: createElement(component),
+      }),
+    )
+  })
+
```

Comment:
Add a react shortcode that inserts the React component into the example site, this will blow up if the react component tries to run DOM APIs on the server